core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -56,7 +59,7 @@ oppia.factory('UrlService', ['$window', function($window) {
           var currentFieldName = decodeURIComponent(
             queryItems[i].split('=')[0]);
           var currentFieldValue = decodeURIComponent(
-            queryItems[i].split('=')[1]);
+            queryItems[i].split('=')[1]).split('#')[0];
```

Comment:
Do you need this?

Reply:
Yes, because if url is collection_id=0&parent=jaeXmlmvS-jo##input-response-pair-0 in the current state and I go to next state (which is the end), the $location.hash(null) in ConverstaionSkinDirective.js only changes it to collection_id=0&parent=jaeXmlmvS-jo#, so that hash has to be removed for getiing current parentId.
Or should I use $location.search here also?

Reply:
yes, use search/hash/pathname where possible. That way you let the browser handle all the complicated stuff and there's more guarantees on your input values.

Reply:
Ok, will do that then

Reply:
Done, also about the testing, in spec file, I am manually giving the window.search value anyways so that doesn't really check if browser correctly splits the url into search and hash right? Also, the window.search won'y have any hash in it also.